module_generator/README.md

```diff
@@ -0,0 +1,220 @@
+- `namespace` (str): Set the name of the namespace or organization ID in which the module is being created.
+- `resource` (int): Select a resource from a numbered list to be added to the module.
+- `model_name` (str): Set the name of the model of the resource.
+- `initialize_workflow` (bool): Initialize a workflow for building the module and other cloud build info.
+
+## Template Structure
+
+When a module is created using the template, there are a couple of different files created or populated. Some files to note:
+
+- `.viam-gen-info` saves the user inputs, the module creation time, and the generator version.
+- `meta.json` saves the `module_name`, `visibility` (default to public), `language`, `generator_version`, `model`, `api`, and `entrypoint` (default to `run.sh`) into a json file.
+- `src/__main__.py` contains all the functions available to the resource.
+
+---
+
+## Module Creation Example
+
+Create a Python module named `rover` in the namespace `test`. Add a vision service model `vizz`. Do not initialize a github workflow.
+
+```python
+% python3 cookiecutter-template
+  [1/6] module_name (MyModule): rover
+  [2/6] Select language
+    1 - python
+    Choose from [1] (1): 1
+  [3/6] namespace (ns): test
+  [4/6] Select resource
+    1 - arm component
+    2 - base component
+    3 - motor component
+    4 - vision service
+    Choose from [1/2/3/4] (1): 4
+  [5/6] model_name (model): vizz
+  [6/6] initialize_workflow (n): n
+```
+
+### Generated Files
+
+`module_generator/rover/.viam-gen-info`
+
+```json
+{
+  "module_name": "rover",
+  "language": "python",
+  "namespace": "test",
+  "resource": "vision service",
+  "model_name": "vizz",
+  "generator_version": "0.14.0",
+  "generated_on": "2024-08-21 11:50:11"
+}
+```
+
+`module_generator/rover/meta.json`
+
+```json
+{
+  "module_name": "rover",
+  "visibility": "public",
+  "language": "python",
+  "generator": "0.14.0",
+  "models": [
+    {
+      "api": "rdk:service:vision",
+      "model": "test:rover:vizz"
+    }
+  ],
+  "entrypoint": "run.sh"
+}
+```
+
+`module_generator/rover/README.md`
+
+```
+# rover Module
+
+This module implements the Vision API in the `test:rover:vizz` model.
+
+More information on the Vision Service can be found here: https://python.viam.dev/autoapi/viam/services/vision/index.html.
+
+```
+
+`module_generator/rover/src/__main__.py`
+
+This is the raw file right after generation. This newly created file has some missing imports, which will be fixed below.
+
+```python
+import asyncio
+from typing import ClassVar, Mapping, Optional, Self, Sequence, Union
+from viam.module.module import Module
+from viam.proto.app.robot import ComponentConfig
+from viam.proto.common import ResourceName
+from viam.resource.base import ResourceBase
+from viam.resource.easy_resource import EasyResource
+from viam.resource.types import Model, ModelFamily
+from viam.services.vision import Vision
+
+
+class Vizz(Vision, EasyResource):
```

Comment:
We're not inheriting from `Reconfigurable` here but we expose the `reconfigure` method below. This is kind of a tricky case where I think some manual user input will be hard to avoid, _but_ if users define a `reconfigure` method I think it's pretty important to derive from `Reconfigurable`, otherwise `reconfigure_resource` calls won't behave correctly (`reconfigure_resource` checks if a resource `isinstance` of `Reconfigurable`, and I don't believe the `isinstance` method uses duck typing). 

As an aside, might be worth considering getting rid of the `Reconfigurable` type at all and wrapping the call to `reconfigure` in a `try/except` block. but that's a separate conversation and a separate ticket.

Reply:
Should I be removing the `reconfigure` method for now?

Reply:
I looked into `EasyResource`, and this class has a `reconfigure` method. Does that mean that every `EasyResource` potentially doesn't work if a descending class calls this function?